pubsub.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module to handle input from Google Monitoring.
+
+This module defines functions and errors to handle input from Google Monitoring,
+such as Pub/Sub notifications.
+"""
+
+import base64
+import binascii
+
+
+class DataParseError(Exception):
+    """Exception raised for errors in a Pub/Sub notification format.
+
+    Attributes:
+        message: explanation of the error
+    """
+
+    def __init__(self, message):
+        self.message = message
+
+
+def parse_data_from_message(pubsub_received_message):
+    """Parses notification messages from Pub/Sub.
+
+    Args:
+        pubsub_received_message: The JSON message to parse from Pub/Sub.
```

Comment:
In main.py pubsub_received_message is defined as `pubsub_received_message = request.get_json()` so I think it should be in a JSON format. Is there a better way to phrase this?

Reply:
Actually I think I see where I'm misunderstanding, JSON is actually a string syntax which can be converted to a Python dict/list. I'll change this to reflect that.

Reply:
Yeah I think you're right. I played around with printing the types and it looks like request.get_json() will parse the encoded string and return the decoded Python dictionary/list or the original Python string if it can't be converted into a dictionary/list. So you're right that pubsub.parse_data_from_message() should be expecting a Python dictionary in this case.